Bottom: 497aaf231d839f2651d6e05fa9df4323061bb55b
Top:    4fa7b5f8a099ed41bf326dfaa6d95cbbe3d48ae4
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-10 12:19:58 -0500

Configure Docker Daemon


---
diff --git a/roles/docker-common/tasks/configure.yml b/roles/docker-common/tasks/configure.yml
new file mode 100644
index 00000000..c1f016e1
--- /dev/null
+++ b/roles/docker-common/tasks/configure.yml
@@ -0,0 +1,25 @@
+---
+
+- name: Create Docker Configuration Directory
+  file:
+    path: /etc/docker
+    state: directory
+    owner: root
+    group: root
+    mode: 'u=rw,g=r,o=r'
+
+- name: Configure Docker Daemon
+  template: 
+    dest: /etc/docker/daemon.json
+    src: daemon.json.j2
+    owner: root
+    group: root
+    mode: 'u=rw,g=r,o=r'
+  register: docker_json
+
+- name: Restart Docker Daemon
+  service:
+    name: docker
+    enabled: yes
+    state: reloaded
+  when: docker_json.changed
diff --git a/roles/docker-common/tasks/main.yml b/roles/docker-common/tasks/main.yml
index e955efa4..f52391f5 100644
--- a/roles/docker-common/tasks/main.yml
+++ b/roles/docker-common/tasks/main.yml
@@ -5,10 +5,12 @@
   when: ansible_os_family == "RedHat"
 
 - import_tasks: ip_addr.yml
+
 - import_tasks: packages.yml
 - import_tasks: sshd_settings.yml
 - import_tasks: mariadb_settings.yml
 
+- import_tasks: configure.yml
 
 
 - name: Check if baculavirt selinux module exists
diff --git a/roles/docker-common/templates/daemon.json.j2 b/roles/docker-common/templates/daemon.json.j2
new file mode 100644
index 00000000..57de7390
--- /dev/null
+++ b/roles/docker-common/templates/daemon.json.j2
@@ -0,0 +1,4 @@
+{
+  "log-driver": "json-file",
+  "log-opts": {"max-size": "10m", "max-file": "3"}
+}
